MembraneRTC/src/main/java/org/membraneframework/rtc/InternalMembraneRTC.kt

```diff
@@ -412,7 +408,18 @@ constructor(
 
             midToTrackId = sdpAnswer.data.midToTrackId
 
-            pc.setRemoteDescription(answer)
+            pc.setRemoteDescription(answer).onSuccess {
+                drainCandidates()
+            }
+        }
+    }
+
+    private fun drainCandidates() {
+        if (this.queuedRemoteCandidates != null) {
```

Comment:
There is the kotlin synax for exactly this:

```kotlin
this.queuedRemoteCancidates?.let {
  for (candidate in it) {
    this.peerConnection?.addIceCandidate(candidate)
  }
}
this.queuedRemoteCandidates = null

```

Reply:
And also we do we want to change the list to `null` instead of just clearing/reseting it? Also is it possible to get a race condition when adding element to list? Don't we need some kind of a mutex here? (I don't remember if those happen on different threads when dispatching the events)

Reply:
> There is the kotlin synax for exactly this:

And they think it's better than good old java syntax? 

> And also we do we want to change the list to null instead of just clearing/reseting it?

This would break `onRemoteCandidate` function. When we get a remote candidate we need to decide whether to add it to the queue or add it directly to the peer connection. The queue for remote candidates has to live until SDP answer is set. After setting SDP answer we need to delete queue and start adding candidates directly to the peer connection. 

> Also is it possible to get a race condition when adding element to list? Don't we need some kind of a mutex here? (I don't remember if those happen on different threads when dispatching the events)

Indeed, this might be a problem. I have protected `queuedRemoteCandidates` with mutex.

Reply:
> And they think it's better than good old java syntax?
The reason kotlin was created is because of java and its boilerplate, I would follow the kotlin practices whenever possible 😉